Okta/OktaKeychain.swift

```diff
@@ -0,0 +1,143 @@
+            kSecClass as String: kSecClassGenericPassword as String,
+            kSecValueData as String: data,
+            kSecAttrAccount as String: key,
+            kSecAttrAccessible as String: accessibility ?? kSecAttrAccessibleWhenUnlockedThisDeviceOnly
+        ] as [String : Any]
+        
+        if let accessGroup = accessGroup {
+            q[kSecAttrAccessGroup as String] = accessGroup
+        }
+        
+        let cfDictionary = q as CFDictionary
+        // Delete existing (if applicable)
+        SecItemDelete(cfDictionary)
+        
+        let sanityCheck = SecItemAdd(cfDictionary, nil)
+        if sanityCheck != noErr {
+            throw OktaKeychainError.failed(sanityCheck.description)
+        }
+        
+        if let expiration = expiration {
+            DispatchQueue.main.asyncAfter(deadline: .now() + expiration) {
+                try? remove(key: key)
+            }
+        }
+    }
+    
+    /**
+     Retrieve the stored JWK information from the Keychain.
+     - parameters:
+     - key: Hash to reference the stored Keychain item
+     */
+    class func get(key: String) throws -> String {
+        let data: Data = try get(key: key)
+        guard let string = String(data: data, encoding: .utf8) else {
+            throw OktaKeychainError.codingError
+        }
+        return string
+    }
+    
+    /**
+     Retrieve the stored JWK information from the Keychain.
+     - parameters:
+     - key: Hash to reference the stored Keychain item
+     */
+    class func get(key: String) throws -> Data {
+        let q = [
+            kSecClass as String: kSecClassGenericPassword,
+            kSecReturnData as String: kCFBooleanTrue,
+            kSecMatchLimit as String: kSecMatchLimitOne,
+            kSecAttrAccount as String: key
+        ] as CFDictionary
+        
+        var ref: AnyObject? = nil
+        
+        let sanityCheck = SecItemCopyMatching(q, &ref)
+        guard sanityCheck == noErr else {
+            throw OktaKeychainError.failed(sanityCheck.description)
+        }
+        guard let data = ref as? Data else {
+            throw OktaKeychainError.failed("No data for \(key)")
+        }
+        return data
+    }
+    
+    /**
+     Remove the stored JWK information from the Keychain.
+     - parameters:
+     - key: Hash to reference the stored Keychain item
+     */
+    class func remove(key: String) throws {
+        let data: Data = try get(key: key)
+        let q = [
+            kSecClass as String: kSecClassGenericPassword as String,
+            kSecValueData as String: data,
+            kSecAttrAccount as String: key
+        ] as CFDictionary
+        
+        // Delete existing (if applicable)
+        let sanityCheck = SecItemDelete(q)
+        guard sanityCheck == noErr else {
+            throw OktaKeychainError.failed(sanityCheck.description)
+        }
+    }
+    
+    /**
+     Removes all entities from the Keychain.
+     */
+    class func clearAll() {
```

Comment:
Can you create a follow-up task to add tests for this new file?

Reply:
created

Reply:
https://github.com/okta/okta-sdk-appauth-ios/issues/92